common/db/openssl/openssl.go

```diff
@@ -15,9 +16,10 @@ import (
 
 // For connecting to the database over ssl
 type SSLDBConnector struct {
-	dialInfo  *mgo.DialInfo
-	dialError error
-	ctx       *openssl.Ctx
+	dialInfo       *mgo.DialInfo
+	dialErrorMutex sync.Mutex
+	dialError      error
```

Comment:
Despite the chattiness, logging the error is the most correct and useful thing to do currently. It gives per-connection information in the most accurate way.